diffkemp/simpll/Utils.h

```diff
@@ -137,10 +133,12 @@ void findAndReplace(std::string &input, std::string find, std::string replace);
 const Instruction *getConstExprAsInstruction(const ConstantExpr *CEx);
 
 /// Retrives type of the value based its C source code identifier.
-Type *getCSourceIdentifierType(
+std::pair<Type *, const DIType *> getCSourceIdentifierType(
```

Comment:
This feels a little odd - I wonder if it could be rewritten in a better way so that this doesn't return a pair of types, which is confusing.

Reply:
That makes sense to me. Again, it also needs to return the name of the type and its size, so I can reuse the new `TypeInfo` structure.

Reply:
If we can return `TypeInfo`, that would be awesome.

Reply:
In the end, the most natural solution was to get rid of the `Type *` and just return `DIType *`.